Graphics/HLSLTools/src/DXILUtils.cpp

```diff
@@ -0,0 +1,701 @@
+/*
+ *  Copyright 2019-2020 Diligent Graphics LLC
+ *  Copyright 2015-2019 Egor Yusov
+ *  
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *  
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *  
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ *
+ *  In no event and under no legal theory, whether in tort (including negligence), 
+ *  contract, or otherwise, unless required by applicable law (such as deliberate 
+ *  and grossly negligent acts) or agreed to in writing, shall any Contributor be
+ *  liable for any damages, including any direct, indirect, special, incidental, 
+ *  or consequential damages of any character arising as a result of this License or 
+ *  out of the use or inability to use the software (including but not limited to damages 
+ *  for loss of goodwill, work stoppage, computer failure or malfunction, or any and 
+ *  all other commercial damages or losses), even if such Contributor has been advised 
+ *  of the possibility of such damages.
+ */
+
+#include <unordered_set>
+#include <unordered_map>
+#include <memory>
+#include <array>
+
+#ifdef WIN32
+#    include <Unknwn.h>
+#    include <guiddef.h>
+#    include <atlbase.h>
+#    include <atlcom.h>
+#endif
+
+#include "DXILUtils.hpp"
+
+// Platforms that has DXCompiler.
+#if defined(PLATFORM_WIN32) || defined(PLATFORM_UNIVERSAL_WINDOWS) || defined(PLATFORM_LINUX)
+
+#    if D3D12_SUPPORTED
+#        include <d3d12shader.h>
+#    endif
+
+#    include "dxc/dxcapi.h"
+
+#    include "DataBlobImpl.hpp"
+#    include "RefCntAutoPtr.hpp"
+
+namespace Diligent
+{
+namespace
+{
+
+#    ifdef PLATFORM_WIN32
+struct DXCompilerBase
```

Comment:
The definition of DXCompilerBase mixes different platforms in one file, which I am trying to avoid. Can you move DXCompilerBase along with all required includes to an appropriate header (DXCompilerBaseLiunx.hpp, DXCompilerBaseWin32.hpp, etc.) and include those headers here?